docs/_static/js/search.js

```diff
@@ -0,0 +1,269 @@
+    }
+  }
+
+  return [start, end];
+}
+
+function addHeadingForQuery(query, parentElement) {
+  const searchHeading = document.createElement('h1');
+  searchHeading.textContent = `Search results for “${query}”`;
+  parentElement.appendChild(searchHeading);
+}
+
+function createHitElement(hitData, query) {
+  const pageURL = new URL(hitData.url);
+  pageURL.hash = '';
+  pageURL.searchParams.set('highlight', query);
+  const anchorURL = new URL(hitData.url);
+  anchorURL.searchParams.set('highlight', query);
+  const result = hitData._highlightResult; // eslint-disable-line
+
+  const hitListElement = document.createElement('li');
+
+  const hierarchies = Object.values(result.hierarchy);
+  const firstHierarchyLevel = hierarchies[0];
+  const lastHierarchyLevel = hierarchies[hierarchies.length - 1];
+
+  const pageLink = document.createElement('a');
+  hitListElement.appendChild(pageLink);
+  pageLink.innerHTML = firstHierarchyLevel.value;
+  pageLink.href = pageURL;
+
+  const contextElement = document.createElement('div');
+  hitListElement.appendChild(contextElement);
+  contextElement.className = 'context';
+
+  if (lastHierarchyLevel && lastHierarchyLevel !== firstHierarchyLevel) {
+    const contextLinkContainer = document.createElement('div');
+    contextElement.appendChild(contextLinkContainer);
+    const contextLink = document.createElement('a');
+    contextLinkContainer.appendChild(contextLink);
+    contextLink.innerHTML = lastHierarchyLevel.value;
+    contextLink.href = anchorURL;
+  }
+
+  if (result.content) {
+    const contentElement = document.createElement('div');
+    contentElement.innerHTML = result.content.value;
+    contextElement.appendChild(contentElement);
+  }
+
+  return hitListElement;
+}
+
+function addResultsList(hits, query, parentElement) {
+  const searchResultsList = document.createElement('ul');
+  searchResultsList.className = 'search';
+  hits.forEach((hit) => {
+    const hitElement = createHitElement(hit, query);
+    searchResultsList.appendChild(hitElement);
+  });
+  parentElement.appendChild(searchResultsList);
+}
+
+function runSearchPageSearch(page) {
+  const urlParams = new URLSearchParams(window.location.search);
+  const query = urlParams.get('q');
+
+  const searchResultsContainer = document.getElementById('search-results');
+
+  // Erase previous results
+
+  if (searchResultsContainer) {
+    searchResultsContainer.innerHTML = '';
+    document.querySelector('.pagination-list').innerHTML = '';
+    addHeadingForQuery(query, searchResultsContainer);
+  }
+
+  const docSearch = docSearchReady();
+  const index = docSearch.client.initIndex('wagtail');
+
+  index
+    .search(query, {
+      hitsPerPage: 100,
+      page: page,
+      facetFilters: [getVersionFacetFilter()],
+    })
+    .then((result) => {
+      // Display pagination if more than 1 page returned
+      const { nbPages } = result;
+      if (nbPages > 1) {
+        document.querySelector('#pagination').hidden = false;
```

Comment:
This can’t work because we’ve removed `id="pagination"` from that element. There’s lots of other code below which also relies on this id, so I’d assume the pagination is completely broken as-is.